File: src/FoodSharing.Site/Models/Announcements/AnnouncementStatistics.cs
namespace FoodSharing.Site.Models.Announcements;

public class AnnouncementStatistics
{
    public Guid AnnouncementId { get; }
    public Int32 ViewsCount { get; }
    public Int32 FavoriteCount { get; }
    public Int32 MessageCount { get; }

    public AnnouncementStatistics(Guid announcementId, Int32 viewsCount, Int32 favoriteCount, Int32 messageCount)
    {
        AnnouncementId = announcementId;
        ViewsCount = viewsCount;
        FavoriteCount = favoriteCount;
        MessageCount = messageCount;
    }
}

File: src/FoodSharing.Site/ClientApp/src/domain/announcements/announcementStatistics.ts
export class AnnouncementStatistics {
    constructor(
        public announcementId: string,
        public viewsCount: number,
        public favoriteCount: number,
        public messageCount: number

    ) { }
}

export function mapToAnnouncementStatistics(data: any) {
    return new AnnouncementStatistics(
        data.announcementId,
        data.viewsCount,
        data.favoriteCount,
        data.messageCount
    )
}